src/YUILoader.cc

```diff
@@ -156,6 +157,35 @@ void YUILoader::loadUI( bool withThreads )
     }
 }
 
+void YUILoader::loadTestFrameworkPlugin( )
+{
+        // Do not try to login if test framework is not enabled
+        if( !getenv( YUITest_ENABLE )  && strcmp(getenv( YUITest_ENABLE ), "1") != 0 ) return;
+        yuiMilestone () << "Requested to start http server to control UI." << std::endl;
+
+        if( pluginExists( YUIPlugin_Test ) ) {
+                YUIPlugin uiPlugin( YUIPlugin_Test );
+                if ( uiPlugin.success() )
+                {
+                        yuiMilestone () << "Loading http server to control UI." << std::endl;
+                        typedef void (*getServerFunction_t)();
+                        getServerFunction_t getServer = (getServerFunction_t) uiPlugin.locateSymbol( "getServer" );
+                        // Error will be already reported by locateSymbol if method not found
+                        if ( getServer )
+                                getServer();
+
+                }
+                else
+                {
+                        yuiError () << "Loading http server to control UI failed." << std::endl;
```

Comment:
Both error conditions in this function might better be exceptions; if you explicitly requested the test plugin in your setup, it is IMHO a fatal condition if it cannot be loaded. That should fail early and with a well-defined error, otherwise the failure might not be detected in time. 

This will very likely lead to a lot of extra work once the guru knowledge how this test plugin works is not widespread anymore (which means basically from next week on). ;-)

Reply:
I was not 100% sure about what to do in such case and though more about keeping it safe and not failing installation.